File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/access/ProvideItems.java
package com.lasthopesoftware.bluewater.client.browsing.items.access;

import com.lasthopesoftware.bluewater.client.browsing.items.Item;
import com.namehillsoftware.handoff.promises.Promise;

import java.util.List;

public interface ProvideItems {

	Promise<List<Item>> promiseItems(int itemKey);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/access/ProvideItems.kt
package com.lasthopesoftware.bluewater.client.browsing.items.access

import com.lasthopesoftware.bluewater.client.browsing.items.IItem
import com.lasthopesoftware.bluewater.client.browsing.items.ItemId
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.namehillsoftware.handoff.promises.Promise

interface ProvideItems {
    fun promiseItems(libraryId: LibraryId, itemId: ItemId? = null): Promise<List<IItem>>
}
